components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/IdentityOathEventListener.java

```diff
@@ -130,7 +120,7 @@ public boolean doPostSetUserClaimValue(String userName, UserStoreManager userSto
         }
         return revokeTokensOfLockedUser(userName, userStoreManager) &&
                 revokeTokensOfDisabledUser(userName, userStoreManager)
-                && removeUserClaimsFromCache(userName, userStoreManager);
+                && OAuthUtil.removeUserClaimsFromCache(userName, userStoreManager);
```

Comment:
Isn't this happening in IdentityOauthEventHandler?

Reply:
IdentityOauthEventHandler listenn only to  POST_SET_USER_CLAIMS. not POST_SET_USER_CLAIM